### PR TITLE
feat(build): Set head-ref in non-PR GitHub workflows (EME-654)

### PR DESCRIPTION
**Pull request by runningcode**
## Summary
- Extends `get_github_head_ref()` to support non-PR GitHub Actions workflows
- For PR workflows: continues to use `GITHUB_HEAD_REF` (unchanged)
- For non-PR workflows (push, release, etc.): now uses `GITHUB_REF_NAME`
- Adds comprehensive tests for all workflow types

## Changes
Modified `src/utils/vcs.rs`:
- Updated `get_github_head_ref()` to handle non-PR events by reading `GITHUB_REF_NAME`
- Added test coverage for push workflows, feature branches, tags, and edge cases

## Impact
Build uploads in non-PR workflows (like Android release builds) will now automatically have the correct branch or tag reference set, without requiring manual `--head-ref` flags.

Fixes EME-654

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/utils/vcs.rs [hunk 1]
```diff
@@ -1585,6 +1590,51 @@ mod tests {
         std::env::remove_var("GITHUB_EVENT_NAME");
     }
 
+    #[test]
+    #[serial(github_env)]
```
Comment (runningcode) [inline src/utils/vcs.rs:1594]: This is set to serial because we're setting github env vars (like other tests that also do this). We can also split up these tests but I figured it saves all the cleanup and teardown before and after every test if we do it all together.
